src/Files.Sdk.Storage/Extensions/StorageExtensions.Service.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Files.Sdk.Storage.LocatableStorage;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Files.Sdk.Storage.Extensions
+{
+	public static partial class StorageExtensions
+	{
+		/// <summary>
+		/// Checks whether the directory exists at a given path and retrieves the folder, otherwise retrieves the file.
+		/// </summary>
+		/// <param name="storageService">The service.</param>
+		/// <param name="path">Path to get the storable at.</param>
+		/// <param name="cancellationToken">A <see cref="CancellationToken"/> that cancels this action.</param>
+		/// <returns>A <see cref="Task"/> that represents the asynchronous operation. Value is <see cref="ILocatableStorable"/> that represents the item.</returns>
+		public static async Task<ILocatableStorable> GetStorableFromPathAsync(this IStorageService storageService, string path, CancellationToken cancellationToken = default)
+		{
+			if (await storageService.DirectoryExistsAsync(path, cancellationToken))
+			{
+				return await storageService.GetFolderFromPathAsync(path, cancellationToken);
+			}
+			else
+			{
+				return await storageService.GetFileFromPathAsync(path, cancellationToken);
+			}
```

Comment:
`Else` is not needed.

Reply:
I think in this case, it makes things a bit more staight-forward